src/utils.js

```diff
@@ -64,8 +64,38 @@ const utils = {
     hasSingleItemOf(arr, itemValue) {
         return arr.length === 1
             && (arguments.length === 2 ? arr[0] === itemValue : true);
-    }
+    },
+
+    isArrIndex (note) {
+        return /^\[\d+\]$/.test(note);
+    },
+
+    splitNotation (notation) {
```

Comment:
Should this be handled?
`x = 'a["prop[]"]'`

`splitNotation(x)`

will return

```
[
'a',
'["prop['
']"'
]
```